hypex/dataset/dataset.py

```diff
@@ -319,6 +320,41 @@ def _check_other_dataset(self, other):
             raise ConcatDataError(type(other))
         if type(other._backend) is not type(self._backend):
             raise ConcatBackendError(type(other._backend), type(self._backend))
+        
+    def astype(
+        self,
+        dtype: Dict[str, DefaultRoleTypes],
+        errors: Literal["raise", "ignore"] = "raise"
+    ) -> "Dataset":
+        """
+        Change the data type of one or more columns.
+        
+        Parameters:
+        - dtype: Dictionary where keys are column names and values are target types.
+        - errors: If 'raise', raises an error on invalid types; if 'ignore', skips invalid types.
+        
+        Returns:
+        - A new Dataset object with the specified data types applied.
+        """
+        
+        for col, _ in dtype.items():
+            if col not in self.columns:
+                if errors == "raise":
+                    raise KeyError(f"Column '{col}' does not exist in the Dataset.")
+
+        new_backend = deepcopy(self._backend)
+        new_backend.data = new_backend.astype(dtype, errors)
+        new_roles = deepcopy(self.roles)
+        
+        if errors == "ignore":
+            for col, target_type in dtype.items():
+                if new_backend.get_column_type(col) == target_type:
+                    new_roles[col].data_type = target_type
+        elif errors == 'raise':
+            for col, target_type in dtype.items():
+                new_roles[col].data_type = target_type
```

Comment:
either you don't need this condition at all, since the error should have been already raised by the backend atype operation, or you mannually should raise it in case the cast in the backend didnt't happen.

Reply:
```PY
for col, target_type in dtype.items():
    if (errors == "raise") or (errors == "ignore" and new_backend.get_column_type(col) == target_type):
        new_roles[col].data_type = target_type
```